File: vueproject1/src/shared/api/types/product.ts
import type { BaseEntity } from './types.ts';
import type { Image } from './images'
import type { Badge } from './badge.ts'
import type {ProductGroup } from './productGroup'


export interface Product extends BaseEntity {
  name: string;
  subName?: string;
  description: string;
  images?: Image[];
  price?: number;
  quantity?: number;
  badges?: Badge[];
  groups?: ProductGroup[];
}

File: Data/Product.cs
namespace Data
{
    public class Product:BaseEntity
    {
        public string Name { get; set; } = "";
        public string SubName { get; set; } = "";
        public string Description { get; set; } = "";
        public virtual List<Image>? Images { get; set; }
        public decimal Price { get; set; }
        public int Quantity { get; set; }
        public virtual List<Badge>? Badges { get; set; }
        public virtual List<ProductGroup>? Groups { get; set; }
    }
}
